src/Umbraco.Web/Runtime/WebRuntimeComponent.cs

```diff
@@ -278,7 +278,10 @@ private static void ConfigureClientDependency(IGlobalSettings globalSettings)
                 { "compositeFileHandlerPath", ClientDependencySettings.Instance.CompositeFileHandlerPath }
             });
 
-            ClientDependencySettings.Instance.MvcRendererCollection.Add(renderer);
+            // When using a non-web runtime and this component is loaded ClientDependency explodes because it'll
+            // want to access HttpContext.Current, which doesn't exist
+            if (HttpContext.Current != null)
```

Comment:
@aaronpowell is there an HttpContext.Current at this stage in the startup cycle? Otherwise `HostingEnvironment.IsHosted` could be used.

Reply:
Ah, as it turns out there's a `IOHelper.IsHosted` method which checks for both of these things that could be used instead.

Reply:
Swapped to `IOHelper.IsHosted`